File: src/simulation/TranscriptGenerator.java
package simulation;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import Debt_LRA_Transcript.Transcript;
import Enums.LetterGrade;
import data.DataManager;
import lecture.Lecture;
import lecture.Schedule;
import lecture.Semester;
import person.Student;

public class TranscriptGenerator {

	private Map<String, Integer> termAndYear;

	public TranscriptGenerator() {

	}

	public Transcript generate(Student student, Schedule schedule) {
		build();
		Transcript transcript = new Transcript(student, null);
		
		transcript.setListOfSemester(setAllSemester(student, schedule));

		return student.getTranscript();
	}

	private void build() {
		termAndYear = new HashMap<String, Integer>();
		termAndYear.put("FallFreshman", 0);
		termAndYear.put("SpringFreshman", 1);
		termAndYear.put("FallSophomore", 2);
		termAndYear.put("SpringSophomore", 3);
		termAndYear.put("FallJunior", 4);
		termAndYear.put("SpringJunior", 5);
		termAndYear.put("FallSenior", 6);
		termAndYear.put("SpringSenior", 7);
	}

	private LetterGrade randomLetterGrade() {

		LetterGrade grade = LetterGrade.None;
		switch ((int) (Math.random() * 9)) {
		case 0:
			grade = LetterGrade.FF;
			break;
		case 1:
			grade = LetterGrade.FD;
			break;
		case 2:
			grade = LetterGrade.DD;
			break;
		case 3:
			grade = LetterGrade.DC;
			break;
		case 4:
			grade = LetterGrade.CC;
			break;
		case 5:
			grade = LetterGrade.CB;
			break;
		case 6:
			grade = LetterGrade.BB;
			break;
		case 7:
			grade = LetterGrade.BA;
			break;
		case 8:
			grade = LetterGrade.AA;
			break;
		}
		return grade;
	}

	private Semester semesterGenerator(Student student, Schedule schedule, int i) {

		Semester semester = new Semester(null);

		Map<Lecture, LetterGrade> listOfLecture = new HashMap<Lecture, LetterGrade>();
		for (Lecture l : DataManager.getInstance().searchLecture(i)) {
			if (student.canTakeLecture(l, student.getTranscript()))
				listOfLecture.put(l, randomLetterGrade());
		}
		semester.setListOfLecturesTaken(listOfLecture);

		return semester;
	}

	private List<Semester> setAllSemester(Student student, Schedule schedule) {
		List<Semester> semesterList = new ArrayList<Semester>();

		for (int i = 0; i < termAndYear.get(schedule.getTerm() + "" + schedule.getTermYear()); i++) {
			semesterList.add(semesterGenerator(student, schedule, i));
		}

		return semesterList;
	}

}

File: pythonCode/TranscriptGenerator.py
from LectureRegistrationApplication import LectureRegistrationApplication
from LetterGrade import LetterGrade
from DataManager import DataManager
from Lecture import Lecture
from Schedule import Schedule
from Semester import Semester
from Student import Student
import random
from TermAndYear import TermAndYear


class TranscriptGenerator():



    def __init__(self):
        pass

    def generate(self, student, schedule):
        #transcript = Transcript()
        transcript.setListOfSemester(self.setAllSemester(student, schedule))
        return transcript

    def __randomLetterGrade(self):
        
        a = random.randint(0, 9)
        grade = LetterGrade.Non
        if a==0:
            grade = LetterGrade.FF
        elif a==1:
            grade = LetterGrade.FD
        elif a==2:
            grade = LetterGrade.DD
        elif a==3:
            grade = LetterGrade.DC
        elif a==4:
            grade = LetterGrade.CC
        elif a==5:
            grade = LetterGrade.CB
        elif a==6:
            grade = LetterGrade.BB
        elif a==7:
            grade = LetterGrade.BA
        elif a==8:
            grade = LetterGrade.AA
        return grade

    def semesterGenerator(self, student, schedule, i):
        """ generated source for method semesterGenerator """
        semester = Semester()
        listOfLecture = {hash(Lecture): LetterGrade for Lecture, LetterGrade in listOfLecture.items()}
        for l in DataManager.getInstance().searchLecture(i):
            if student.getTranscript() == None:
                listOfLecture.put(l, self.randomLetterGrade())
                continue 
            if student.canTakeLecture(l, student.getTranscript()):
                listOfLecture.put(l, self.randomLetterGrade())
        semester.setListOfLecturesTaken(listOfLecture)
        return semester

    def setAllSemester(self, student, schedule):
        """ generated source for method setAllSemester """
        semesterList = List()
        i = 0
        while i < TermAndYear[schedule.Term+""+schedule.TermYear].value:
            semesterList.add(self.semesterGenerator(student, schedule, i))
            i += 1
        return semesterList